src/machine/machine_atmega328p.go

```diff
@@ -467,3 +472,93 @@ var SPI0 = SPI{
 	sdo:  PB3,
 	sdi:  PB4,
 	cs:   PB2}
+
+// Pin Change Interrupts
+var (
+	pinCallbacks [3][8]func(Pin)
+	interrupts   [3]*interrupt.Interrupt
+	previous     [3]uint8
+)
+
+func handlePCINTInterrupts(intr uint8, port *volatile.Register8) {
+	current := port.Get()
+	for i := uint8(0); i < 8; i++ {
+		if (current>>i)&0x01 == 0x01 && pinCallbacks[intr][i] != nil {
+			pinCallbacks[intr][i](Pin(i * (intr + 1)))
+		}
+	}
+	previous[intr] = current
+}
+
+func handlePCINT0Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT0-avr.IRQ_PCINT0, avr.PINB)
+}
+
+func handlePCINT1Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT1-avr.IRQ_PCINT0, avr.PINC)
+}
+
+func handlePCINT2Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT2-avr.IRQ_PCINT0, avr.PIND)
+}
+
+func SetInterrupt(pin Pin, callback func(Pin)) (err error) {
+	if callback == nil {
+		// Disable this pin interrupt
+		return ErrFeatureNotImplemented
+	}
+	var PCMSK_REG *volatile.Register8
+	var PCIE uint8
+	var PCINT uint8
+	var IRQ uint8
+	var PIN0 Pin
+
+	switch {
+	case pin >= PB0 && pin <= PB7:
+		// PCMSK0 - PCINT0-7
+		PCMSK_REG = avr.PCMSK0
+		PCIE = avr.PCICR_PCIE0
+		PIN0 = PD0
+		PCINT = 1 << (pin - PIN0)
+		IRQ = avr.IRQ_PCINT0
```

Comment:
You could consider the following:

```suggestion
		pinIndex := pin - PD0
		pinCallbacks[0][pinIndex] = callback
		avr.PCMSK0.SetBits(1 << pinIndex)
		avr.PCICR.SetBits(avr.PCICR_PCIE0)
```

It's less abstract, but it's actually a bit shorter and perhaps more readable as well. Also, I think the compiler will be better able to optimize this code.

Reply:
Yeah, I was intentionally being abstract because... well... reasons.
However, it ended up not working out very well that way due to... other reasons. 
This new way simplifies quite a bit so I'll have the MR updated soon.

...I also need to figure out why "make release" is now crashing my machine....

Reply:
So, I've `make release`, at least to the point that it runs to completion by ripping out all of tinygo and golang and reinstalling them.
However, I now get
```
$ tinygo build -target=arduino   
warning: no avr-gcc installation can be found on the system, cannot link standard libraries
warning: standard library not linked and so no interrupt vector table or compiler runtime routines will be linked
warning: no avr-gcc installation can be found on the system, cannot link standard libraries
warning: standard library not linked and so no interrupt vector table or compiler runtime routines will be linked
```
However,
```
avr-gcc
avr-gcc: fatal error: no input files
compilation terminated.
```

Reply:
Please see https://tinygo.org/getting-started/install/linux/#avr-eg-arduino-uno for the install requirements for AVR on Linux.